source/agora/crypto/Hash.d

```diff
@@ -120,31 +118,29 @@ public Hash hashFull (T) (scope const auto ref T record)
 }
 
 /// Ditto
-public void hashPart (T) (scope const auto ref T record, scope HashDg state)
-    /*pure*/ nothrow @nogc
+public void hashPart (T) (scope const auto ref T record, scope HashDg hasher)
+    /*pure*/ nothrow @nogc @trusted
```

Comment:
Why not `@safe` ?

Reply:
Note that normally, attributes are inferred on templates, so we usually don't put any, unless we want to ensure they don't get changed in a way that breaks those attributes.

Reply:
Because of the cast

Reply:
I added inner function with trusted in else block. Slightly better?